cpp/oneapi/dal/backend/common.hpp

```diff
@@ -416,7 +458,7 @@ inline std::int64_t device_max_sg_size(const sycl::queue& q) {
 inline std::int64_t propose_wg_size(const sycl::queue& q) {
     // TODO: a temporary solution that limits work item count used on the device.
     // Needs to change to more smart logic in the future.
-    return std::min<std::int64_t>(512, device_max_wg_size(q));
+    return std::min<std::int64_t>(1024, device_max_wg_size(q));
```

Comment:
As far as I understand this change affects not only KMeans++.
Have you checked that the performance of other algorithms has not degraded after the change?

Reply:
I have checked only for a limited amount of algorithms. It slightly improved performance. 

Anyway the rule of thumb is that bigger work groups should result in better performance.